src/unix/stream.c

```diff
@@ -394,7 +394,8 @@ int uv__stream_open(uv_stream_t* stream, int fd, int flags) {
 #if defined(__APPLE__)
   enable = 1;
   if (setsockopt(fd, SOL_SOCKET, SO_OOBINLINE, &enable, sizeof(enable)) &&
-      errno != ENOTSOCK) {
+      errno != ENOTSOCK &&
+      errno != EINVAL) {
```

Comment:
I think ENOPROTOOPT is returned only if the level was incorrectly set, and SOL_SOCKET is the correct one for this option.

```
[ENOPROTOOPT]      The option is unknown at the level indicated.
```